app/src/main/java/com/endava/myendava/activities/SignInActivity.java

```diff
@@ -57,22 +86,43 @@ private void setupModule() {
     @Override
     protected void onResume() {
         super.onResume();
-        mSignInButton.setOnClickListener(v -> showMainScreen());
+        mSignInButton.setOnClickListener(v -> validateEmail());
         mSignInAsGuestButton.setOnClickListener(v -> showSignInAsGuestScreen());
+        mForgotPasswordButton.setOnClickListener(v -> handleForgotPassword());
     }
 
+
     private void showSignInAsGuestScreen() {
         SignInAsGuestActivity.start(this);
         finish();
     }
 
-    private void showMainScreen() {
+    private void handleSignIn() {
+        if (!isMailValid) {
+            Toast.makeText(this, INVALID_EMAIL_MESSAGE, Toast.LENGTH_SHORT).show();
+            return;
+        }
         mSharedPreferences.setUserAsEmployee();
         mSharedPreferences.setUserEmail(mUserEmailEditText.getText().toString());
         MainActivity.start(this);
         finish();
     }
 
+    private void handleForgotPassword() {
+        Toast.makeText(this, "AHA", Toast.LENGTH_SHORT).show();
+    }
+
+    private void validateEmail() {
+        String email = mUserEmailEditText.getText().toString();
+        if (email.length() > 0) {
+            mUsersViewModel.checkUserEmail(email).observe(this, valid -> {
+                this.isMailValid = valid.booleanValue();
+                handleSignIn();
+            });
+        } else Toast.makeText(this, INVALID_EMAIL_MESSAGE, Toast.LENGTH_SHORT).show();
```

Comment:
always put brackets to if and else for better visibility ... no brackets is a C/C++ coding style only